### PR TITLE
feat: 🎸 image endpoint

### PR DESCRIPTION
**Pull request by thehairy**

### CODE REVIEW

File: pages/api/image.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+declare module 'formidable' {
+    interface FileJSON {
+        newFilename: string;
+    }
+}
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+    if (req.method === "POST") {
+        const form = new IncomingForm({
```
Comment (thehairy) [inline pages/api/image.ts:15]: Have to think about a different way on handling those images.
I cant save them to the server with Netlify/Vercel.

File: pages/api/image.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                    if (fields?.token === process.env.API_SECRET) {
+                        const { origin } = absoluteUrl(req);
+                        return res.status(200).json({
+                            message: 'success',
+                            url: `${origin}/${file.toJSON().newFilename}`,
+                        });
+                    } else {
+                        file.destroy();
+                        return res.status(401).json({
+                            error: 'You are not authorized for this endpoint. Please provide a valid token or get support at \'https://imageing.org/discord\'.',
```
Comment (tristancamejo) [inline pages/api/image.ts:38]: You should make some error class to pass a code (enum) and message.

File: pages/api/image.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                        });
+                    }
+                } else {
+                    console.log(files)
+                    return res.status(500).json({ error: 'There has been an issue on server side. Please try again or get support at \'https://imageing.org/discord\'.' });
```
Comment (tristancamejo) [inline pages/api/image.ts:43]: You should make some error class to pass a code (enum) and message.

File: pages/api/image.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                }
+            } else {
+                return res.status(40).json({ error: 'Provide a file to the request body. Please try again or get support at \'https://imageing.org/discord\'.' });
```
Comment (tristancamejo) [inline pages/api/image.ts:46]: You should make some error class to pass a code (enum) and message.

**Review by tristancamejo — CHANGES_REQUESTED** [review]
LGTM, just address requests.

**Comment by thehairy** [comment]
Will move the api to a different repo and host it myself.

**Pull request closed without merging**